### PR TITLE
Gowda cr

### PR DESCRIPTION
**Pull request by MSG17**
**Program Analysis**
Technically works, although with rather minimal progress made currently.

**Was the program available in UC Github on time?**
Yes, it was available and in a runnable state.

**Is the program documented/commented well enough for you to understand?**
Although most of the program was relatively straightforward, the lack of comments really obfuscated the functionality of certain functions and methods.

**Does the program compile?**
Yes, the program builds, compiles and runs in the emulator. However, after allowing location use, it crashed every time I tried to open it until I disabled location perms.

**Change rationale**
*Technical:*
Updated MockK to work with functionality used in unit testing so that InstitutionDataIntegrationTest would pass
Took out duplication in MainFragment (one of the seven deadly sins of development, [see here](https://edwin.vriethoff.net/2016/08/25/improve-code-quality-getting-started-sonarqube/)) by instilling a new function for changing the visibility of UI elements
Also removed unused imports in MainFragment
*Documentation and formatting* (done according to [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html)):
Added documentation for DTO classes
Corrected whitespace (eg `){` to `) {` for declarations) and camelcase

**What technical aspects did I learn about?**
Requesting user permissions
How to use the device's camera and gather photo metadata
Use of MockK in unit tests and mocking in general

### CODE REVIEW

**Review by MSG17 — COMMENTED** [review]
Added documentation to DTO classes.

File: app/src/main/java/com/example/ranchratings_12/dtos/Institution.kt [hunk 1]
```diff
@@ -1,4 +1,8 @@
 package com.example.ranchratings_12.dtos
 
-class Institution (val name: String, val streetAddress: String, val phoneNumber: String){
+/**
+ * Stores details on the restaurant or other institution reviewed.
```
Comment (discospiff) [inline app/src/main/java/com/example/ranchratings_12/dtos/Institution.kt:4]: Proper KDoc should describe parameters, with @param annotation, and return values, with @return annotation.

File: app/src/main/java/com/example/ranchratings_12/dtos/LocationDetails.kt [hunk 2]
```diff
@@ -1,5 +1,8 @@
 package com.example.ranchratings_12.dtos
 
+/**
+ * Stores coordinates of the institution location for the GPS.
```
Comment (discospiff) [inline app/src/main/java/com/example/ranchratings_12/dtos/LocationDetails.kt:4]: See comments above.

File: app/src/main/java/com/example/ranchratings_12/dtos/Review.kt [hunk 3]
```diff
@@ -1,4 +1,8 @@
 package com.example.ranchratings_12.dtos
 
-data class Review (var userID: Int = 0, var institutionName: String = "", var latitude: String = "", var longitutde: String = "", var reviewText: String = "", var rating: Double = 0.0){
+/**
+ * Stores the review itself, detailing the user, location, institution and rating.
```
Comment (discospiff) [inline app/src/main/java/com/example/ranchratings_12/dtos/Review.kt:4]: See comments above.

File: app/src/main/java/com/example/ranchratings_12/ui/main/MainFragment.kt [hunk 4]
```diff
@@ -65,47 +62,41 @@ class MainFragment : Fragment() {
...
+        btnTakePhoto.visibility = review
+        btnSave.visibility = review
+        txtLatitude.visibility = review
+        txtLongitude.visibility = review
+        txtAddReview.visibility = home
+        btnAddReview.visibility = home
+        txtInstitutionName.visibility = review
+        btnSearch.visibility = home
+        btnProfile.visibility = home
+    }
```
Comment (discospiff) [inline app/src/main/java/com/example/ranchratings_12/ui/main/MainFragment.kt:98]: Hmmm, ok.  Nice consolidation.

File: app/src/test/java/com/example/ranchratings_12/InstitutionDataIntegrationTest.kt [hunk 5]
```diff
@@ -15,7 +15,7 @@ import org.junit.rules.TestRule
  *
  * See [testing documentation](http://d.android.com/tools/testing).
  */
-class InstitutionDataintegrationTest {
+class InstitutionDataIntegrationTest {
```
Comment (discospiff) [inline app/src/test/java/com/example/ranchratings_12/InstitutionDataIntegrationTest.kt:18]: Proper UpperCamelCase naming.

File: app/src/test/java/com/example/ranchratings_12/InstitutionDataUnitTest.kt [hunk 6]
```diff
@@ -55,7 +55,7 @@ class InstitutionDataUnitTest {
     }
 
     private fun createMockData() {
-        var allinstitutionsLiveData = MutableLiveData<ArrayList<Institution>>()
+        var allInstitutionsLiveData = MutableLiveData<ArrayList<Institution>>()
```
Comment (discospiff) [inline app/src/test/java/com/example/ranchratings_12/InstitutionDataUnitTest.kt:58]: Proper naming.  Good.

**Comment by discospiff** [comment]
Several good suggestions here.  I'd fully implement the JavaDoc, and merge.

File: app/src/main/java/com/example/ranchratings_12/dtos/Institution.kt (hunk 1 above)
Comment (MSG17) [reply app/src/main/java/com/example/ranchratings_12/dtos/Institution.kt]: According to [Kotlin's convention on comments](https://kotlinlang.org/docs/coding-conventions.html#documentation-comments), `@param` and `@return` notation should be avoided. Nevertheless, the params should still be marked as `[param]` in comments, which I have added in my last commit.

**Pull request merged**